controlplane/kubeadm/controllers/status.go

```diff
@@ -66,6 +67,12 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 		return nil
 	}
 
+	machinesWithHealthAPIServer := ownedMachines.Filter(collections.HealthyAPIServer())
```

Comment:
Why is this based on only healthy machines instead of all machines in the current control plane

Reply:
If we look at machines only there is the risk that we report the control plane version before the control plane actually exists; by considering only machines with healthy api servers we avoid this problem and we are also reflecting more accurately the actual state of the control plane

Reply:
If we look at machines only there is the risk that we report the control plane version before the control plane actually exists; by considering only machines with healthy api servers we avoid this problem and we are also reflecting more accurately the actual state of the control plane